### PR TITLE
Pin SciPy version to latest; regenerate `scipy.special` signatures

### PR DESCRIPTION
**Pull request by person142**
Closes https://github.com/numba/numba-scipy/issues/35.

At this early point in the project I don't think it's worth trying to
support multiple versions of SciPy simultaneously; this could easily
overwhelm limited developer time better spent adding new
features. Instead this pins SciPy at the latest version (1.4.1), the
idea being that `numba-scipy` and `scipy` releases are done in
lockstep.

It's easy to relax this in the future if more resources are available.

### CODE REVIEW

**Comment by person142** [comment]
I understand if this solution isn't palatable, but I figured I would throw it out there. It feels higher impact to work on things like casting/array input support/complex number support for special functions at this point over maintaining internal whitelists/blacklists.

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 numba
-scipy
+scipy==1.4.1
```
Comment (stuartarchibald) [inline requirements.txt:2]: This should should also go into the conda build meta.yaml?

**Comment by rileymcdowell** [comment]
I really like the speedup I get by using numba-scipy. Any idea of when we can get a release with the new 1.4.1 bindings and the scipy version pinned?

**Comment by stuartarchibald** [comment]
@rileymcdowell once https://github.com/numba/numba-scipy/pull/37#pullrequestreview-387153374 is resolved I think a new release can be cut.

**Comment by person142** [comment]
I will try to get back to this over the weekend.

**Comment by person142** [comment]
> Should this bonding to a SciPy version be so specific, i.e. patch version point

Pinned to 1.4.* instead.

> Would it be worth making the overload/import mechanism more tolerant instead so as to either correctly bind to a pointer if it's there, or raise an error message if used and it's not ?

Yes, did that. Now this will happen:

- If a function in our big list isn't found in SciPy at import time, no overloads will be generated for it
- If a signature is requested for a function and it isn't present in the available signatures, then an error will be raised at the time of request

With these changes I can successfully use the signatures generated for 1.4 with 1.3 (as long as the function/signature also exists in 1.3).

**Comment by person142** [comment]
CI still needs some work, but I'm out of time for the day.

**Comment by grburgess** [comment]
thanks for this. I installed this branch and now everything is working great again

**Comment by person142** [comment]
Ok, CI has been appeased. The matrix needed some version bumping since NumPy 1.15 is pretty old now.

**Comment by person142** [comment]
It seems like SciPy hasn't been built for 32 bit Linux since 1.1?

https://anaconda.org/anaconda/scipy

I removed it from the matrix; happy to add it back if it's actually available.

**Comment by stuartarchibald** [comment]
@person142 Thanks for doing this and fighting CI! Given there's now some capacity in the registration mech to handle "missing/changed stuff", I'm wondering if the strong bind to SciPy=1.14 can be relaxed, it'd be nice to keep a 32bit build running for RPi users?

**Comment by person142** [comment]
> I'm wondering if the strong bind to SciPy=1.14 can be relaxed

Sure, we’d just need to start skipping tests conditionally on old SciPy versions.

**Comment by bdch1234** [comment]
Any idea on when this will be merged? Current release of numba-scipy seems to be pretty much broken from what I can tell, see this e.g. https://github.com/numba/numba-scipy/issues/44

Another example is this I am getting:
```
UserWarning: Numba extension module 'numba_scipy' failed to load due to 'ValueError(No function '__pyx_fuse_0pdtr' found in __pyx_capi__ of 'scipy.special.cython_special')'.
  entrypoints.init_all()
```
This is using latest scipy/numba/numba-scipy.

**Comment by HDembinski** [comment]
Is this project dead? Because I need the functionality and am considering a fork.

**Comment by stuartarchibald** [comment]
> Is this project dead? Because I need the functionality and am considering a fork.

Unfortunately, Numba, like a lot of open source projects, has limited resources. The buildfarm that builds Numba etc. has been undergoing a large overhaul and the graphs for building this haven't been restored yet. I think given this is a pure python package it could be moved to github actions, we'll try and do that shortly (next week or so) to coincide with the 0.53RC series for Numba. Thanks for your understanding :)

**Comment by stuartarchibald** [comment]
@esc fixed being able to build and ship in https://github.com/numba/numba-scipy/pull/47, thanks @esc :)

**Comment by esc** [comment]
> @esc fixed being able to build and ship in #47, thanks @esc :)

No problem, at least we can ship dev versions again now. The missing component is still to ship to PyPi when we tag, but that is a bridge we can cross when we get to it.

**Comment by esc** [comment]
I wonder if we can run GHA against this? It might need an upstream merge from `master` and a bump of the SciPy versions in the GHAs?

**Comment by esc** [comment]
@person142 can you merge `master` into this branch and push? I think it should trigger the Github Actions.

**Comment by person142** [comment]
Done... though now SciPy 1.4 is a bit long in the tooth-there aren't wheels for 3.9 e.g.

**Comment by esc** [comment]
@person142 thank you for doing that. It looks like the github actions did run now, but seem to have failed:


```
E       numba.core.errors.TypingError: No implementation of function Function(<ufunc 'wrightomega'>) found for signature:
```

**Comment by esc** [comment]
Note: I am pretty sure that this PR will be abandoned in favor of #54 -- please don't waste time or energy working on it, thank you. Also, @person142 thank you again for your initial and continued efforts to move this forward, we really appreciate your efforts!

**Comment by esc** [comment]
Closing this now as https://github.com/numba/numba-scipy/pull/54 has been merged.

**Pull request closed without merging**